apps/studio/src/server/modules/folder/folder.router.ts

```diff
@@ -220,4 +265,58 @@ export const folderRouter = router({
         return pick(result, defaultFolderSelect)
       },
     ),
+
+  getIndexpage: protectedProcedure
+    .input(getIndexpageSchema)
+    .query(async ({ input: { resourceId, siteId } }) => {
+      // TODO: add permissions checking
+
+      return db.transaction().execute(async (tx) => {
+        const { title } = await tx
+          .selectFrom("Resource")
+          .where("id", "=", resourceId)
+          .select("title")
+          .executeTakeFirstOrThrow()
+
+        let indexPage = await tx
+          .selectFrom("Resource")
+          .where("Resource.parentId", "=", resourceId)
+          .where("Resource.type", "=", "IndexPage")
+          .select(["id", "draftBlobId"])
+          .executeTakeFirst()
+
+        // NOTE: Might want to create on click in
+        // rather than via the view
+        if (!indexPage) {
+          const indexPageBlob = await tx
+            .insertInto("Blob")
+            .values({
+              content: jsonb(createIndexPage(title)),
+            })
+            .returning("id")
+            .executeTakeFirstOrThrow()
+
+          indexPage = await tx
+            .insertInto("Resource")
+            .values({
+              parentId: resourceId,
+              draftBlobId: indexPageBlob.id,
+              title,
+              type: ResourceType.IndexPage,
+              permalink: INDEX_PAGE_PERMALINK,
+              siteId,
+            })
+            .returning(["id", "draftBlobId"])
+            .executeTakeFirstOrThrow()
+
```

Comment:
issue: I think we are missing an audit log entry here?

Reply:
added!